### PR TITLE
Update HistoryCommand and CommandHistory to be more History-like #534

### LINKED ISSUE #534: Update HistoryCommand to be more History-like
The HistoryCommand returns elements in chronological order.

This goes against the intuition that history ought to be seen in reverse chronological order.

### PR DESCRIPTION
**Pull request by Zhiyuan-Amos**
Fixes #534

### CODE REVIEW

**Comment by CanIHasReview** [comment]
# v1

@Zhiyuan-Amos submitted v1 for review.

* [v1 1/2] [CommandHistory: Update API to be more History\-like](https://github.com/se-edu/addressbook-level4/pull/535/commits/87630ea900c42ab08f2d7fce976cb5f61e417457)
* [v1 2/2] [HistoryCommand: Return user input in reverse chronological order](https://github.com/se-edu/addressbook-level4/pull/535/commits/fea1de19fce52012abf61c938c13ec0ceada9b71)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/daddc4289462285c4474311539b79287465273cc...fea1de19fce52012abf61c938c13ec0ceada9b71))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/535/1/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

File: src/main/java/seedu/address/logic/CommandHistory.java [hunk 1]
```diff
@@ -16,11 +17,11 @@ public CommandHistory() {
     }
 
     /**
-     * Appends {@code userInput} to the list of user input entered.
+     * Prepends {@code userInput} to the list of user input entered.
```
Comment (yamgent) [inline src/main/java/seedu/address/logic/CommandHistory.java]: `Adds {@code userInput}...`

**Comment by yamgent** [comment]
As spoken, `CommandHistory` need not be stored in a reverse chronological order. However, the `HistoryCommand` will have to reverse the `CommandHistory` and print it in a reverse chronological order.

So it is as follows:

```
CommandHistory

  +-------------+------+---------------+
  |least recent |  abc | most recent   |
  +-------------+------+---------------+
  
Index:   0           1           2
```

```
HistoryCommand's result display:

Entered commands (from earliest to most recent):
most recent
abc
least recent
```

File: src/main/java/seedu/address/logic/CommandHistory.java (hunk 1 above)
Comment (limmlingg) [inline src/main/java/seedu/address/logic/CommandHistory.java]: Prepend implies adding (at the front of the list)

File: src/main/java/seedu/address/logic/CommandHistory.java [hunk 2]
```diff
@@ -16,11 +17,11 @@ public CommandHistory() {
...
      */
     public void add(String userInput) {
         requireNonNull(userInput);
-        userInputHistory.add(userInput);
+        userInputHistory.add(0, userInput);
```
Comment (limmlingg) [inline src/main/java/seedu/address/logic/CommandHistory.java]: Remove magic number. Maybe `START_INDEX`?

**Comment by Zhiyuan-Amos** [comment]
@yamgent you discussed with prof? Also, I'm of the opinion that `CommandHistory` should be of reverse chronological order because it's history-related.

**Comment by Zhiyuan-Amos** [comment]
Closing this PR following comment made in: https://github.com/se-edu/addressbook-level4/pull/491#issuecomment-312266122

**Pull request closed without merging**